### DISCUSSION TITLE
Technical differences between SignalR streams and "finalized" data

### DISCUSSION THREAD

**Discussion by pesaventofilippo** [theOehrly/Fast-F1#456] [General]
Session data from [livetiming.formula1.com](https://livetiming.formula1.com/) is usually available about 30 minutes after a session ends, but all data is transmitted live with signalr (and we can record it with the `fastf1.livetiming` module).

If you look at [SessionInfo.json](https://livetiming.formula1.com/static/SessionInfo.json) (the data for the latest session), `ArchiveStatus` is set to `Generating` while the session is in progress until 30 minutes after the session, then it becomes `Archived` and we can pull the data from the APIs.

My question is, what additional data (if any) is being generated or anyway provided after the session ends, that was not already present in the signalr broadcast?
If I hypothetically record every single message broadcasted with signalr from start to finish, could I parse & compute the same data that is later available from the APIs? Or is there some other reason we can't just use signalr data directly?

**Comment by theOehrly** [comment]
It should be exactly the same. But the timestamps differ between the SignalR stream and the archive that becomes available later. So you can't mix the two. The archives may cut off some unimportant data at the beginning and the end.
The best guess for why it takes so long is that the recording simply runs until the session is finalized (suggestion of multiviewer dev) .
